pkg/sql/pgx/pgx_adapter.go

```diff
@@ -0,0 +1,128 @@
+package pgx
+
+import (
+	"context"
+	stdSQL "database/sql"
+	"fmt"
+	"time"
+
+	"github.com/ThreeDotsLabs/watermill-sql/v3/pkg/sql"
+
+	"github.com/jackc/pgconn"
+	"github.com/jackc/pgx/v4"
+)
+
+type Conn interface {
+	BeginTx(ctx context.Context, options pgx.TxOptions) (pgx.Tx, error)
+	Exec(ctx context.Context, sql string, arguments ...interface{}) (pgconn.CommandTag, error)
+	Query(ctx context.Context, sql string, arguments ...interface{}) (pgx.Rows, error)
+}
+
+type Beginner struct {
+	Conn
+}
+
+type Tx struct {
+	pgx.Tx
+}
+
+type Result struct {
+	pgconn.CommandTag
+}
+
+type Rows struct {
+	pgx.Rows
+}
+
+func (c Beginner) BeginTx(ctx context.Context, options *stdSQL.TxOptions) (sql.Tx, error) {
+	opts := pgx.TxOptions{}
+	if options != nil {
+		iso, err := toPgxIsolationLevel(options.Isolation)
+		if err != nil {
+			return nil, err
+		}
+
+		opts = pgx.TxOptions{
+			IsoLevel:       iso,
+			AccessMode:     toPgxAccessMode(options.ReadOnly),
+			DeferrableMode: "",
+		}
+	}
+
+	tx, err := c.Conn.BeginTx(ctx, opts)
+
+	return Tx{tx}, err
+}
+
+func (c Beginner) ExecContext(ctx context.Context, query string, args ...interface{}) (sql.Result, error) {
+	res, err := c.Conn.Exec(ctx, query, args...)
+
+	return Result{res}, err
+}
+
+func (c Beginner) QueryContext(ctx context.Context, query string, args ...interface{}) (sql.Rows, error) {
+	rows, err := c.Conn.Query(ctx, query, args...)
+
+	return Rows{rows}, err
+}
+
+func (t Tx) ExecContext(ctx context.Context, query string, args ...any) (sql.Result, error) {
+	res, err := t.Tx.Exec(ctx, query, args...)
+
+	return Result{res}, err
+}
+
+func (t Tx) QueryContext(ctx context.Context, query string, args ...any) (sql.Rows, error) {
+	rows, err := t.Tx.Query(ctx, query, args...)
+
+	return Rows{rows}, err
+}
+
+func (t Tx) Rollback() error {
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+
+	return t.Tx.Rollback(ctx)
+}
+
+func (t Tx) Commit() error {
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+
+	return t.Tx.Commit(ctx)
+}
```

Comment:
Because the stdSQL interface this interface uses doesn't take a context, it looks like I decided to create a context and add a timeout here. 

I don't want to refactor too much to make context available here in this PR.

1. I could just use a background context with no timeout
2. I could make the timeout configurable when creating this adapter
3. I could keep a hardcoded default timeout as it is currently (I don't like this as much)